feat: Add Acala/Karura logo, update bufferMultiplier

zjb0807 opened the pull request:
## **Description**

Adding token logos and network info for Acala/Karura

Set Acala/Karura/AcalaTestnet `bufferMultiplier` to 1 in `CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP`


## **Related issues**

Fixes:

## **Manual testing steps**


1. Add Acala Mainnet and check if it appears with logo and expected information
2. Transfer to contract address, the `gasLimit` is correct.

3. Add Karura Mainnet and check if it appears with logo and expected information.
4. Transfer to contract address, the `gasLimit` is correct.

5. On Acala Testnet, transfer to the contract address, the `gasLimit` is correct.

## **Screenshots/Recordings**

### **Before**

![image](https://github.com/user-attachments/assets/0ca64868-3c0f-41e3-9351-5de254da72ea)

![image](https://github.com/user-attachments/assets/a29e5aff-8eb6-4879-8ae5-d6cc5d2e1feb)
![image](https://github.com/user-attachments/assets/ce183ae2-fedb-40c2-bea9-25777e61db0c)

### **After**

![image](https://github.com/user-attachments/assets/15d23d36-91d6-46de-9178-aebeca28dc8e)
![image](https://github.com/user-attachments/assets/0fc83368-0346-4318-b808-6c4900c7cf87)
![image](https://github.com/user-attachments/assets/9f962e9f-50d2-4f50-af1a-7029800e447f)


## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

zjb0807 committed:
chore: Add Acala/Karura logo, update CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP

zjb0807 commented:
I have read the CLA Document and I hereby sign the CLA

zjb0807 committed:
trigger GitHub actions

zjb0807 committed:
Merge branch 'MetaMask:main' into add-acala-karura-mandala

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
Update display name

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
resolve conflicts

zjb0807 commented:
> @zjb0807 I don't know anything about `bufferMultiplier`. Do you know why `CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP` is 1 only for `OPTIMISM` and `OPTIMISM_TESTNET` right now, and why you need it for 3 new networks?

Because we have 3 networks. 
The Karura is the parachain of Kusama, and Acala is the parachain of Polkadot. `ACALA_MANDALA_TESTNET` is a testnet.

https://github.com/MetaMask/metamask-extension/pull/11435 It seems that `OPTIMISM` and `OPTIMISM_TESTNET` do not need buffer either.

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 commented:
@gambinish @HowardBraham Sorry to bother you. 
It would be great if you could help me submit an internal PR.

HowardBraham committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
Apply suggestions

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 commented:
@gambinish @HowardBraham  Hi team, I'm not sure why CI fails. Would you happen to have an idea about this?

HowardBraham commented:
> @gambinish @HowardBraham Is the TimeoutError here due to the changes made in this PR? Is there anything missing?

Ah, I think it might be failing for all external contributors.  I also tried to get the attention of someone who knew more about  parachains and bufferMultiplier, but I wasn't successful.

zjb0807 commented:
@HowardBraham 

I just added 3 network information, updated IMAGE_URL and CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP, there should be no security issues

> Ah, I think it might be failing for all external contributors. I also tried to get the attention of someone who knew more about parachains and bufferMultiplier, but I wasn't successful.

@HowardBraham Thank you for your efforts on my PR. Would you please help me submit an internal PR to avoid the CI failure issue?

I just added 3 network information, updated `IMAGE_URL` and `CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP`, there should be no security issues.

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

zjb0807 commented:
Hi @HowardBraham, I saw your #35320 should fix the CI. Can we move forward with this PR?

HowardBraham commented on shared/constants/network.ts:
```diff
@@ -1263,6 +1281,9 @@ export const ETHERSCAN_SUPPORTED_NETWORKS = {
 export const CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP = {
   [CHAIN_IDS.OPTIMISM]: 1,
   [CHAIN_IDS.OPTIMISM_TESTNET]: 1,
+  [CHAIN_IDS.ACALA]: 1,
+  [CHAIN_IDS.ACALA_TESTNET]: 1,
+  [CHAIN_IDS.KARURA]: 1,
```
@zjb0807 I just tested it myself, and I'm not sure these lines are necessary.  But I saw you wrote a little about manual testing and added screenshots.  Can you show me a very specific example that makes these lines necessary?

zjb0807 replied:
We have encoded the storage fee data in `gasLimit`. If it is changed, the transaction will fail. We have increased the `gasLimit` before encoding, so the transaction will not fail with the returned gas.

zjb0807 committed:
Merge branch 'main' into add-acala-karura-mandala

itsyoboieltr reviewed (approved):
LGTM!
